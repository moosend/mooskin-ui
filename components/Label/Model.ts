import { IBoxLabelComponentProps } from '../Box/model';

export interface ILabelComponentProps extends IBoxLabelComponentProps {
	disabled?: boolean;
	limit?: number;
}
