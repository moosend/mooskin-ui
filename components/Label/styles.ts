import styled from 'styled-components';

// Models
import { ILabelComponentProps } from './model';

// Components
import { Box } from '../Box/Box';

// "CSS" variables
import variables from '../_utils/globals/variables';

export const StyledLabel = styled(Box)<ILabelComponentProps>`
	font-size: 14px;
	font-weight: 500;
	display: flex;
	color: ${(props) => props.palette?.fontColors.text || variables.fontColors.text};
	cursor: ${(props) => (!props.disabled ? (props.onClick ? 'pointer' : 'unset') : 'not-allowed')};
`;

StyledLabel.displayName = 'StyledLabel';
