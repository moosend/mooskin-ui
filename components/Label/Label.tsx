import * as React from 'react';

// Mooskin Context HoC that passes context to component props
import { withMooskinContext } from '../Styled/MooskinContextProvider';

// Models
import { ILabelComponentProps } from './model';

// Styled Components
import { StyledLabel } from './styles';

/**
 * Label
 */
export const Label: React.FC<ILabelComponentProps> = withMooskinContext((props) => {
	const onClick = (e: React.MouseEvent<HTMLLabelElement>) => {
		!props.disabled && props.onClick && props.onClick(e);
	};

	const children =
		props.limit && typeof props.children === 'string'
			? props.children && props.children.length > props.limit
				? props.children.substr(0, props.limit) + '...'
				: props.children
			: props.children;

	return <StyledLabel {...props} onClick={props.onClick ? onClick : undefined} children={children} boxAs="label" />;
});

Label.defaultProps = {
	className: '',
	style: {},
};

Label.displayName = 'Label';
