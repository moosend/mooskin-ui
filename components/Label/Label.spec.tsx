import * as React from 'react';
import { Label } from './Label';

import { mount } from 'enzyme';

describe('Label', () => {
	test('renders correctly', () => {
		const tree = mount(<Label>Label here!</Label>);
		expect(tree).toMatchSnapshot();
	});
});
