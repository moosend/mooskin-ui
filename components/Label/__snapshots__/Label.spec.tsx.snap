// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label renders correctly 1`] = `
<Label
  className=""
  style={Object {}}
>
  <Component
    className=""
    palette={
      Object {
        "backgroundColors": Object {
          "altWhite": "#FFFFFF",
          "background": "#FAFAFA",
          "black": "#000000",
          "checkboxUnselected": "rgba(0, 0, 0, 0.62)",
          "darkgray1": "#777777",
          "darkgray2": "#454545",
          "darkgray3": "#424242",
          "dropdown": "#FFFFFF",
          "gray1": "#E0E0E0",
          "gray100": "#F0F0F0",
          "gray2": "#E7E7E7",
          "grayAlpha": "rgba(0,0,0,0.17)",
          "inverseText": "#212121",
          "medgray1": "#757575",
          "medgray2": "#888888",
          "primary1": "#4169E1",
          "primary2": "#212121",
          "primary3": "#00796B",
          "primary4": "#D32F2F",
          "secondary1": "#2E7D32",
          "secondary2": "#757575",
          "secondary3": "#673AB7",
          "secondary4": "#9C27B0",
          "secondary5": "#00796B",
          "secondary6": "#ECF0FC",
          "text": "#212121",
          "white": "#FFFFFF",
        },
        "borderColors": Object {
          "altWhite": "#FFFFFF",
          "background": "#FAFAFA",
          "black": "#000000",
          "checkboxUnselected": "rgba(0, 0, 0, 0.62)",
          "darkgray1": "#777777",
          "darkgray2": "#454545",
          "darkgray3": "#424242",
          "dropdown": "#FFFFFF",
          "gray1": "#E0E0E0",
          "gray2": "#E7E7E7",
          "grayAlpha": "rgba(0,0,0,0.17)",
          "grayAlpha100": "rgba(0,0,0,0.06)",
          "inverseText": "#212121",
          "medgray1": "#757575",
          "medgray2": "#888888",
          "primary1": "#4169E1",
          "primary2": "#212121",
          "primary3": "#00796B",
          "primary4": "#D32F2F",
          "secondary1": "#2E7D32",
          "secondary2": "#757575",
          "secondary3": "#673AB7",
          "secondary4": "#9C27B0",
          "secondary5": "#00796B",
          "secondary6": "#ECF0FC",
          "text": "#212121",
          "white": "#FFFFFF",
        },
        "fontColors": Object {
          "altWhite": "#FFFFFF",
          "background": "#FAFAFA",
          "black": "#000000",
          "checkboxUnselected": "rgba(0, 0, 0, 0.62)",
          "darkgray1": "#777777",
          "darkgray2": "#454545",
          "darkgray3": "#424242",
          "dropdown": "#FFFFFF",
          "gray1": "#E0E0E0",
          "gray2": "#E7E7E7",
          "grayAlpha": "rgba(0,0,0,0.17)",
          "grayAlpha100": "rgba(0,0,0,0.06)",
          "grayAlpha400": "rgba(0,0,0,0.36)",
          "inverseText": "#212121",
          "medgray1": "#757575",
          "medgray2": "#888888",
          "primary1": "#4169E1",
          "primary2": "#212121",
          "primary3": "#00796B",
          "primary4": "#D32F2F",
          "secondary1": "#2E7D32",
          "secondary2": "#757575",
          "secondary3": "#673AB7",
          "secondary4": "#9C27B0",
          "secondary5": "#00796B",
          "secondary6": "#ECF0FC",
          "text": "#212121",
          "white": "#FFFFFF",
        },
      }
    }
    style={Object {}}
  >
    <StyledLabel
      boxAs="label"
      className=""
      palette={
        Object {
          "backgroundColors": Object {
            "altWhite": "#FFFFFF",
            "background": "#FAFAFA",
            "black": "#000000",
            "checkboxUnselected": "rgba(0, 0, 0, 0.62)",
            "darkgray1": "#777777",
            "darkgray2": "#454545",
            "darkgray3": "#424242",
            "dropdown": "#FFFFFF",
            "gray1": "#E0E0E0",
            "gray100": "#F0F0F0",
            "gray2": "#E7E7E7",
            "grayAlpha": "rgba(0,0,0,0.17)",
            "inverseText": "#212121",
            "medgray1": "#757575",
            "medgray2": "#888888",
            "primary1": "#4169E1",
            "primary2": "#212121",
            "primary3": "#00796B",
            "primary4": "#D32F2F",
            "secondary1": "#2E7D32",
            "secondary2": "#757575",
            "secondary3": "#673AB7",
            "secondary4": "#9C27B0",
            "secondary5": "#00796B",
            "secondary6": "#ECF0FC",
            "text": "#212121",
            "white": "#FFFFFF",
          },
          "borderColors": Object {
            "altWhite": "#FFFFFF",
            "background": "#FAFAFA",
            "black": "#000000",
            "checkboxUnselected": "rgba(0, 0, 0, 0.62)",
            "darkgray1": "#777777",
            "darkgray2": "#454545",
            "darkgray3": "#424242",
            "dropdown": "#FFFFFF",
            "gray1": "#E0E0E0",
            "gray2": "#E7E7E7",
            "grayAlpha": "rgba(0,0,0,0.17)",
            "grayAlpha100": "rgba(0,0,0,0.06)",
            "inverseText": "#212121",
            "medgray1": "#757575",
            "medgray2": "#888888",
            "primary1": "#4169E1",
            "primary2": "#212121",
            "primary3": "#00796B",
            "primary4": "#D32F2F",
            "secondary1": "#2E7D32",
            "secondary2": "#757575",
            "secondary3": "#673AB7",
            "secondary4": "#9C27B0",
            "secondary5": "#00796B",
            "secondary6": "#ECF0FC",
            "text": "#212121",
            "white": "#FFFFFF",
          },
          "fontColors": Object {
            "altWhite": "#FFFFFF",
            "background": "#FAFAFA",
            "black": "#000000",
            "checkboxUnselected": "rgba(0, 0, 0, 0.62)",
            "darkgray1": "#777777",
            "darkgray2": "#454545",
            "darkgray3": "#424242",
            "dropdown": "#FFFFFF",
            "gray1": "#E0E0E0",
            "gray2": "#E7E7E7",
            "grayAlpha": "rgba(0,0,0,0.17)",
            "grayAlpha100": "rgba(0,0,0,0.06)",
            "grayAlpha400": "rgba(0,0,0,0.36)",
            "inverseText": "#212121",
            "medgray1": "#757575",
            "medgray2": "#888888",
            "primary1": "#4169E1",
            "primary2": "#212121",
            "primary3": "#00796B",
            "primary4": "#D32F2F",
            "secondary1": "#2E7D32",
            "secondary2": "#757575",
            "secondary3": "#673AB7",
            "secondary4": "#9C27B0",
            "secondary5": "#00796B",
            "secondary6": "#ECF0FC",
            "text": "#212121",
            "white": "#FFFFFF",
          },
        }
      }
      style={Object {}}
    >
      <Box
        boxAs="label"
        className="sc-gtssRu gGjztn"
        palette={
          Object {
            "backgroundColors": Object {
              "altWhite": "#FFFFFF",
              "background": "#FAFAFA",
              "black": "#000000",
              "checkboxUnselected": "rgba(0, 0, 0, 0.62)",
              "darkgray1": "#777777",
              "darkgray2": "#454545",
              "darkgray3": "#424242",
              "dropdown": "#FFFFFF",
              "gray1": "#E0E0E0",
              "gray100": "#F0F0F0",
              "gray2": "#E7E7E7",
              "grayAlpha": "rgba(0,0,0,0.17)",
              "inverseText": "#212121",
              "medgray1": "#757575",
              "medgray2": "#888888",
              "primary1": "#4169E1",
              "primary2": "#212121",
              "primary3": "#00796B",
              "primary4": "#D32F2F",
              "secondary1": "#2E7D32",
              "secondary2": "#757575",
              "secondary3": "#673AB7",
              "secondary4": "#9C27B0",
              "secondary5": "#00796B",
              "secondary6": "#ECF0FC",
              "text": "#212121",
              "white": "#FFFFFF",
            },
            "borderColors": Object {
              "altWhite": "#FFFFFF",
              "background": "#FAFAFA",
              "black": "#000000",
              "checkboxUnselected": "rgba(0, 0, 0, 0.62)",
              "darkgray1": "#777777",
              "darkgray2": "#454545",
              "darkgray3": "#424242",
              "dropdown": "#FFFFFF",
              "gray1": "#E0E0E0",
              "gray2": "#E7E7E7",
              "grayAlpha": "rgba(0,0,0,0.17)",
              "grayAlpha100": "rgba(0,0,0,0.06)",
              "inverseText": "#212121",
              "medgray1": "#757575",
              "medgray2": "#888888",
              "primary1": "#4169E1",
              "primary2": "#212121",
              "primary3": "#00796B",
              "primary4": "#D32F2F",
              "secondary1": "#2E7D32",
              "secondary2": "#757575",
              "secondary3": "#673AB7",
              "secondary4": "#9C27B0",
              "secondary5": "#00796B",
              "secondary6": "#ECF0FC",
              "text": "#212121",
              "white": "#FFFFFF",
            },
            "fontColors": Object {
              "altWhite": "#FFFFFF",
              "background": "#FAFAFA",
              "black": "#000000",
              "checkboxUnselected": "rgba(0, 0, 0, 0.62)",
              "darkgray1": "#777777",
              "darkgray2": "#454545",
              "darkgray3": "#424242",
              "dropdown": "#FFFFFF",
              "gray1": "#E0E0E0",
              "gray2": "#E7E7E7",
              "grayAlpha": "rgba(0,0,0,0.17)",
              "grayAlpha100": "rgba(0,0,0,0.06)",
              "grayAlpha400": "rgba(0,0,0,0.36)",
              "inverseText": "#212121",
              "medgray1": "#757575",
              "medgray2": "#888888",
              "primary1": "#4169E1",
              "primary2": "#212121",
              "primary3": "#00796B",
              "primary4": "#D32F2F",
              "secondary1": "#2E7D32",
              "secondary2": "#757575",
              "secondary3": "#673AB7",
              "secondary4": "#9C27B0",
              "secondary5": "#00796B",
              "secondary6": "#ECF0FC",
              "text": "#212121",
              "white": "#FFFFFF",
            },
          }
        }
        style={Object {}}
        transition="0.2s all"
      >
        <Component
          boxAs="label"
          className="sc-gtssRu gGjztn"
          palette={
            Object {
              "backgroundColors": Object {
                "altWhite": "#FFFFFF",
                "background": "#FAFAFA",
                "black": "#000000",
                "checkboxUnselected": "rgba(0, 0, 0, 0.62)",
                "darkgray1": "#777777",
                "darkgray2": "#454545",
                "darkgray3": "#424242",
                "dropdown": "#FFFFFF",
                "gray1": "#E0E0E0",
                "gray100": "#F0F0F0",
                "gray2": "#E7E7E7",
                "grayAlpha": "rgba(0,0,0,0.17)",
                "inverseText": "#212121",
                "medgray1": "#757575",
                "medgray2": "#888888",
                "primary1": "#4169E1",
                "primary2": "#212121",
                "primary3": "#00796B",
                "primary4": "#D32F2F",
                "secondary1": "#2E7D32",
                "secondary2": "#757575",
                "secondary3": "#673AB7",
                "secondary4": "#9C27B0",
                "secondary5": "#00796B",
                "secondary6": "#ECF0FC",
                "text": "#212121",
                "white": "#FFFFFF",
              },
              "borderColors": Object {
                "altWhite": "#FFFFFF",
                "background": "#FAFAFA",
                "black": "#000000",
                "checkboxUnselected": "rgba(0, 0, 0, 0.62)",
                "darkgray1": "#777777",
                "darkgray2": "#454545",
                "darkgray3": "#424242",
                "dropdown": "#FFFFFF",
                "gray1": "#E0E0E0",
                "gray2": "#E7E7E7",
                "grayAlpha": "rgba(0,0,0,0.17)",
                "grayAlpha100": "rgba(0,0,0,0.06)",
                "inverseText": "#212121",
                "medgray1": "#757575",
                "medgray2": "#888888",
                "primary1": "#4169E1",
                "primary2": "#212121",
                "primary3": "#00796B",
                "primary4": "#D32F2F",
                "secondary1": "#2E7D32",
                "secondary2": "#757575",
                "secondary3": "#673AB7",
                "secondary4": "#9C27B0",
                "secondary5": "#00796B",
                "secondary6": "#ECF0FC",
                "text": "#212121",
                "white": "#FFFFFF",
              },
              "fontColors": Object {
                "altWhite": "#FFFFFF",
                "background": "#FAFAFA",
                "black": "#000000",
                "checkboxUnselected": "rgba(0, 0, 0, 0.62)",
                "darkgray1": "#777777",
                "darkgray2": "#454545",
                "darkgray3": "#424242",
                "dropdown": "#FFFFFF",
                "gray1": "#E0E0E0",
                "gray2": "#E7E7E7",
                "grayAlpha": "rgba(0,0,0,0.17)",
                "grayAlpha100": "rgba(0,0,0,0.06)",
                "grayAlpha400": "rgba(0,0,0,0.36)",
                "inverseText": "#212121",
                "medgray1": "#757575",
                "medgray2": "#888888",
                "primary1": "#4169E1",
                "primary2": "#212121",
                "primary3": "#00796B",
                "primary4": "#D32F2F",
                "secondary1": "#2E7D32",
                "secondary2": "#757575",
                "secondary3": "#673AB7",
                "secondary4": "#9C27B0",
                "secondary5": "#00796B",
                "secondary6": "#ECF0FC",
                "text": "#212121",
                "white": "#FFFFFF",
              },
            }
          }
          style={Object {}}
          transition="0.2s all"
        >
          <StyledBox
            as="label"
            boxAs="label"
            className="sc-gtssRu gGjztn"
            palette={
              Object {
                "backgroundColors": Object {
                  "altWhite": "#FFFFFF",
                  "background": "#FAFAFA",
                  "black": "#000000",
                  "checkboxUnselected": "rgba(0, 0, 0, 0.62)",
                  "darkgray1": "#777777",
                  "darkgray2": "#454545",
                  "darkgray3": "#424242",
                  "dropdown": "#FFFFFF",
                  "gray1": "#E0E0E0",
                  "gray100": "#F0F0F0",
                  "gray2": "#E7E7E7",
                  "grayAlpha": "rgba(0,0,0,0.17)",
                  "inverseText": "#212121",
                  "medgray1": "#757575",
                  "medgray2": "#888888",
                  "primary1": "#4169E1",
                  "primary2": "#212121",
                  "primary3": "#00796B",
                  "primary4": "#D32F2F",
                  "secondary1": "#2E7D32",
                  "secondary2": "#757575",
                  "secondary3": "#673AB7",
                  "secondary4": "#9C27B0",
                  "secondary5": "#00796B",
                  "secondary6": "#ECF0FC",
                  "text": "#212121",
                  "white": "#FFFFFF",
                },
                "borderColors": Object {
                  "altWhite": "#FFFFFF",
                  "background": "#FAFAFA",
                  "black": "#000000",
                  "checkboxUnselected": "rgba(0, 0, 0, 0.62)",
                  "darkgray1": "#777777",
                  "darkgray2": "#454545",
                  "darkgray3": "#424242",
                  "dropdown": "#FFFFFF",
                  "gray1": "#E0E0E0",
                  "gray2": "#E7E7E7",
                  "grayAlpha": "rgba(0,0,0,0.17)",
                  "grayAlpha100": "rgba(0,0,0,0.06)",
                  "inverseText": "#212121",
                  "medgray1": "#757575",
                  "medgray2": "#888888",
                  "primary1": "#4169E1",
                  "primary2": "#212121",
                  "primary3": "#00796B",
                  "primary4": "#D32F2F",
                  "secondary1": "#2E7D32",
                  "secondary2": "#757575",
                  "secondary3": "#673AB7",
                  "secondary4": "#9C27B0",
                  "secondary5": "#00796B",
                  "secondary6": "#ECF0FC",
                  "text": "#212121",
                  "white": "#FFFFFF",
                },
                "fontColors": Object {
                  "altWhite": "#FFFFFF",
                  "background": "#FAFAFA",
                  "black": "#000000",
                  "checkboxUnselected": "rgba(0, 0, 0, 0.62)",
                  "darkgray1": "#777777",
                  "darkgray2": "#454545",
                  "darkgray3": "#424242",
                  "dropdown": "#FFFFFF",
                  "gray1": "#E0E0E0",
                  "gray2": "#E7E7E7",
                  "grayAlpha": "rgba(0,0,0,0.17)",
                  "grayAlpha100": "rgba(0,0,0,0.06)",
                  "grayAlpha400": "rgba(0,0,0,0.36)",
                  "inverseText": "#212121",
                  "medgray1": "#757575",
                  "medgray2": "#888888",
                  "primary1": "#4169E1",
                  "primary2": "#212121",
                  "primary3": "#00796B",
                  "primary4": "#D32F2F",
                  "secondary1": "#2E7D32",
                  "secondary2": "#757575",
                  "secondary3": "#673AB7",
                  "secondary4": "#9C27B0",
                  "secondary5": "#00796B",
                  "secondary6": "#ECF0FC",
                  "text": "#212121",
                  "white": "#FFFFFF",
                },
              }
            }
            style={Object {}}
            transition="0.2s all"
          >
            <label
              className="sc-bdnylx ckoRHe sc-gtssRu gGjztn"
              style={Object {}}
            >
              Label here!
            </label>
          </StyledBox>
        </Component>
      </Box>
    </StyledLabel>
  </Component>
</Label>
`;
